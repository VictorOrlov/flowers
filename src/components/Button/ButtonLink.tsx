import * as React from "react";
import { Link } from "react-router-dom";

interface IProps {
  to: string;
  success?: boolean;
  danger?: boolean;
  check?: boolean;
  children: string | React.ReactNode;
  onClick?: (e: any) => void;
}

const ButtonLink: React.SFC<IProps> = (props: IProps) => {
  const { to, success, danger, check, children, onClick } = props;
  function style() {
    if (success) return "button_success";
    if (danger) return "button_danger";
    if (check) return "button_check";
  }
  return (
    <Link
      className={`buttonLink button_expandet ${style()}`}
      to={to || "/"}
      onClick={onClick}
    >
      <span>{children}</span>
    </Link>
  );
};

ButtonLink.defaultProps = {
  success: false,
  danger: false,
  check: false,
  to: "/"
};

export default ButtonLink;
