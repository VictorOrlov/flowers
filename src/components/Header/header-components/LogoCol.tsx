import * as React from "react";
import { Link } from "react-router-dom";
import logo from "../../../images/gerb.svg";
import logo_text from "../../../images/text.svg";

const LogoCol: React.FC = () => (
  <div className="col-12 col-sm-6 col-md-3">
    <div className="logo_column">
      <Link to="/">
        <div className="logo_svg">
          <img src={logo} alt="logo" />
        </div>
        <div className="logo_text">
          <img src={logo_text} alt="logo text" />
        </div>
      </Link>
    </div>
  </div>
);

export default LogoCol;
