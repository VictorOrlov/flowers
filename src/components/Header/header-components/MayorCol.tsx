import * as React from "react";
import mayor from "../../../images/mayor.jpg";

const MayorCol: React.FC = () => (
  <div className="col-12 col-sm-12 col-md-4 d-none d-sm-block">
    <div className="mayor_column">
      <div className="mayor_card">
        <div className="mayor_img">
          <img src={mayor} alt="mayor" />
        </div>
        <div className="mayor_name">
          <h3>Ильсур</h3>
          <h3>Метшин</h3>
          <h4>Мэр города</h4>
        </div>
      </div>

      <p className="mayor_text">
        Ухоженные деревья, качественные газоны и цветочные композиции стали
        нормой для нашего города и его визитной карточкой. Цветы украшают наш
        город, создавая особое светлое настроение его жителям и гостям.
        Благодаря проекту «Цветущая Казань» город приобрел целостный и
        привлекательный облик цветущего мегаполиса. Отрадно, что к нему
        подключились не только городские предприятия, представители бизнеса, но
        и обычные горожане. В проекте «Цветущая Казань» не может быть
        проигравших. Здесь победителями будем мы все.
      </p>
    </div>
  </div>
);

export default MayorCol;
