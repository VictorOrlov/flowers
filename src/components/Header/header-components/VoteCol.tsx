import * as React from "react";
import ButtonLink from "../../../components/Button/ButtonLink";
import { withRouter } from "react-router";
import { RouteComponentProps } from "react-router-dom";
import { FakeUserType } from "../../../fakeDada/fakeUsers";

interface IProps {
  requestsCount?: React.ReactText;
  voicesCount?: React.ReactText;
  currentUser?: FakeUserType;
}

type VoteProps = IProps & RouteComponentProps;

export const VoteCol: React.SFC<VoteProps> = (props: VoteProps) => {
  const { currentUser, requestsCount, location, voicesCount } = props;
  function showParticBtn() {
    if (currentUser) {
      if (currentUser.participant) {
        return null;
      }
      return (
        <>
          <span className="vote_unit">или</span>
          <div className="new_nomination">
            <ButtonLink
              to="#TakePart"
              danger
              onClick={e => {
                e.preventDefault();
                return window.scrollTo(0, 9000);
              }}
            >
              {" "}
              Участвовать в конкурсе{" "}
            </ButtonLink>
          </div>
        </>
      );
    }
    return (
      <div className="new_nomination">
        <ButtonLink
          to="#TakePart"
          success
          onClick={e => {
            e.preventDefault();
            return window.scrollTo(0, 9000);
          }}
        >
          {" "}
          Принять участие{" "}
        </ButtonLink>
      </div>
    );
  }
  return (
    <div className="col-12 col-sm-6 col-md-5">
      <div className="vote_column">
        <div className="vote">
          <h3 className="vote_title">на текущий момент подано</h3>
          <p>
            <span className="vote_counter">{requestsCount} </span>
            <span className="vote_unit">заявок</span>
          </p>
          <h3 className="vote_title">проголосовало</h3>
          <p>
            <span className="vote_counter">{voicesCount} </span>
            <span className="vote_unit">жителя города</span>
          </p>
          {currentUser && currentUser.remainVoices ? (
            <div className="new_nomination">
              <ButtonLink to="/" success>
                {" "}
                Голосовать{" "}
              </ButtonLink>
            </div>
          ) : null}
          {location.pathname === "/" && (
            <>
              {showParticBtn()}
              <div className="new_nomination">
                <ButtonLink to="/flowerfest" danger>
                  {" "}
                  "Цветочный фестиваль"
                </ButtonLink>
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
};

VoteCol.defaultProps = {
  requestsCount: 0,
  voicesCount: 0
};

export default withRouter(VoteCol);
