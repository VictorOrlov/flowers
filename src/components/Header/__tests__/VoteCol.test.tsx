import React from 'react';
import {VoteCol} from '../header-components/VoteCol';
import { mount } from 'enzyme';
import { 
  UserApproved, UserNotAuth, UserCreated, UserNoParticipation, 
  UserDeclined, UserDeleted, UserDeletedForNewRequest, UserUpdated, UserRemainVoicesEnded 
} from "../../../fakeDada/fakeUsers";
import { StaticRouter } from 'react-router';
import ButtonLink from '../../Button/ButtonLink';


const location = {
  hash: "",
  pathname: "/",
  search: "",
  state: undefined
};
const match = {
  isExact: true,
  params: {},
  path: "/",
  url: "/"
};

const voteColResult = (currentUser: any) => (
  mount(
    <StaticRouter>
      <VoteCol
        requestsCount={1234}
        voicesCount={7777}
        currentUser={currentUser}
        location={location}
        match={match}
      />
    </StaticRouter>
  )
)

it('correct displaing values in VoteCol', ()=>{
  const component = voteColResult(UserNotAuth);
  const requestsCount = component.find('span.vote_counter').at(0).text();
  const voicesCount = component.find('span.vote_counter').at(1).text();
  expect(requestsCount).toBe("1234 ");
  expect(voicesCount).toBe("7777 ");
})

it('render VoteCol Component with user = null', ()=> {
  const component = voteColResult(UserNotAuth);
  expect(component.find(ButtonLink)).toHaveLength(2);
  expect(component.find(ButtonLink).first().text()).toEqual(" Принять участие ");
})

it('render VoteCol Component with user.participant = null',()=> {
  const component = voteColResult(UserNoParticipation)
  expect(component.find(ButtonLink)).toHaveLength(3);
  expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
  expect(component.find(ButtonLink).at(1).text()).toEqual(" Участвовать в конкурсе ")
  expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
});

it('render VoteCol Component with user.remainVoices = 0', ()=> {
  const component = voteColResult(UserRemainVoicesEnded);
  expect(component.find(ButtonLink)).toHaveLength(1);
  expect(component.find(ButtonLink).at(0).text()).toEqual(" \"Цветочный фестиваль\"");
});

describe("render VoteCol Component with ...", ()=> {
  it('... user.participant.request.state = approved', ()=> {
    const component = voteColResult(UserApproved);
    expect(component.find(ButtonLink)).toHaveLength(2);
    expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
    expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
    expect(component.children()).toMatchSnapshot();
  });
  
  it('... user.participant.request.state = created', ()=> {
    const component = voteColResult(UserCreated);
    expect(component.find(ButtonLink)).toHaveLength(2);
    expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
    expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
  });
  
  it('... user.participant.request.state = declined', ()=> {
    const component = voteColResult(UserDeclined);
    expect(component.find(ButtonLink)).toHaveLength(2);
    expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
    expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
  });
  
  it('... user.participant.request.state = deleted', ()=> {
    const component = voteColResult(UserDeleted);
    expect(component.find(ButtonLink)).toHaveLength(2);
    expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
    expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
  });
  
  it('... user.participant.request.state = deleted_for_new_request', ()=> {
    const component = voteColResult(UserDeletedForNewRequest);
    expect(component.find(ButtonLink)).toHaveLength(2);
    expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
    expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
  });
  
  it('... user.participant.request.state = updated', ()=> {
    const component = voteColResult(UserUpdated);
    expect(component.find(ButtonLink)).toHaveLength(2);
    expect(component.find(ButtonLink).last().text()).toEqual(" \"Цветочный фестиваль\"");
    expect(component.find(ButtonLink).first().text()).toEqual(" Голосовать ");
  });
});

it('checking the clickability of the button in VoteCol Component', ()=> {
  const component = voteColResult(UserApproved);
  const a = component.find(ButtonLink).last();
  expect(a.text()).toEqual(" \"Цветочный фестиваль\"");
  expect(a.props().to).toBe('/flowerfest');
});

