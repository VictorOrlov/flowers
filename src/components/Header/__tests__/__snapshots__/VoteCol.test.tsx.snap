// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render VoteCol Component with ... ... user.participant.request.state = approved 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <VoteCol
    currentUser={
      Object {
        "email": "lol.tebeb@n.ru",
        "id": 258,
        "name": "Одобрен",
        "participant": Object {
          "name": "ООО УК \\"ЖКХ ДЕРБЫШКИ\\"",
          "request": Object {
            "address": "Батурина 18п",
            "competition": Object {
              "name": "Лучшие цветы на районе",
            },
            "description": "Почему белые розы похожи на пелемени?",
            "participant": Object {
              "name": "ООО УК \\"ЖКХ ДЕРБЫШКИ\\"",
            },
            "photos": Array [
              Object {
                "address": "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif",
                "id": 1,
              },
              Object {
                "address": "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg",
                "id": 2,
              },
              Object {
                "address": "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg",
                "id": 3,
              },
              Object {
                "address": "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg",
                "id": 4,
              },
            ],
            "state": "approved",
          },
        },
        "phone": "8-987-2222-33-44",
        "remainVoices": 7,
        "roles": Array [
          "member",
        ],
        "surname": "Зубенко",
        "voices": Array [
          117,
          54,
          200,
          693,
        ],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
    requestsCount={1234}
    voicesCount={7777}
  >
    <div
      className="col-12 col-sm-6 col-md-5"
    >
      <div
        className="vote_column"
      >
        <div
          className="vote"
        >
          <h3
            className="vote_title"
          >
            на текущий момент подано
          </h3>
          <p>
            <span
              className="vote_counter"
            >
              1234
               
            </span>
            <span
              className="vote_unit"
            >
              заявок
            </span>
          </p>
          <h3
            className="vote_title"
          >
            проголосовало
          </h3>
          <p>
            <span
              className="vote_counter"
            >
              7777
               
            </span>
            <span
              className="vote_unit"
            >
              жителя города
            </span>
          </p>
          <div
            className="new_nomination"
          >
            <ButtonLink
              check={false}
              danger={false}
              success={true}
              to="/"
            >
              <Link
                className="buttonLink button_expandet button_success"
                to="/"
              >
                <a
                  className="buttonLink button_expandet button_success"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                     
                    Голосовать
                     
                  </span>
                </a>
              </Link>
            </ButtonLink>
          </div>
          <div
            className="new_nomination"
          >
            <ButtonLink
              check={false}
              danger={true}
              success={false}
              to="/flowerfest"
            >
              <Link
                className="buttonLink button_expandet button_danger"
                to="/flowerfest"
              >
                <a
                  className="buttonLink button_expandet button_danger"
                  href="/flowerfest"
                  onClick={[Function]}
                >
                  <span>
                     
                    "Цветочный фестиваль"
                  </span>
                </a>
              </Link>
            </ButtonLink>
          </div>
        </div>
      </div>
    </div>
  </VoteCol>
</Router>
`;
